app/io/pathfinder/controllers/ClusterController.java

```diff
@@ -0,0 +1,103 @@
+package io.pathfinder.controllers;
+
+import com.avaje.ebean.Ebean;
+import com.avaje.ebean.text.json.JsonContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import play.libs.Json;
+import play.mvc.Controller;
+import io.pathfinder.models.Cluster;
+import play.mvc.Result;
+import java.util.Iterator;
+
+import javax.persistence.PersistenceException;
+
+public class ClusterController extends Controller {
+    private JsonContext jsonContext = Ebean.createJsonContext();
+
+    public Result getClusters() {
+        return ok(jsonContext.toJson(Cluster.find.all()));
+    }
+
+    public Result getCluster(long id) {
+        Cluster cluster = Cluster.find.byId(id);
+
+        if (cluster == null) {
+            return notFound();
+        }
+
+        return ok(jsonContext.toJson(cluster));
+    }
+
+    public Result createCluster() {
+        JsonNode json = request().body().asJson();
+        if (json == null) {
+            return badRequest("Expected content-type text/json or application/json");
+        }
+
+        Cluster cluster;
+
+        try {
+            cluster = Json.fromJson(json, Cluster.class);
+
+            cluster.save();
+            return created(jsonContext.toJson(cluster));
+        } catch (PersistenceException e) {
+            return internalServerError("Error saving cluster to the database: " + e.getMessage());
+        } catch (RuntimeException e) {
+            return badRequest("Unable to map json to commodity object: " + e.getMessage());
```

Comment:
From what I have seen, Json.fromJson() is relatively forgiving and will just fill in default values if you don't explicitly give it something. We should take a closer look at what, if any, inputs throw exceptions when mapping from json. I have a feeling that if execution reaches this catch block, it wasn't because of malformed input causing a mapping issue. This might need to return a 500 instead.